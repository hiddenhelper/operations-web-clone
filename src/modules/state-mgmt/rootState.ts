import { History } from 'history';

import { IState as IAuthState } from './auth/state';
import { IState as IGeneralState } from './general/state';
import { IState as IClientState } from './client/state';
import { IState as IProjectState } from './project/state';
import { IState as IUserState } from './user/state';
import { IState as IAccessControlSystemState } from './access-control-system/state';
import { IState as IBadgePrintingSystemState } from './badge-printer-system/state';
import { IState as IWorkerState } from './worker/state';
import { IState as ICertificationState } from './certification/state';
import { IState as IFileState } from './file/state';
import { IState as IBadgeState } from './badge/state';
import { IState as IStatisticsState } from './statistics/state';
import { IState as ITrainingState } from './training/state';
import { IState as IPaymentState } from './payment/state';
import { IState as IInvoiceState } from './invoice/state';
import { ApiService } from '../services/ApiService';
import { AuthService } from '../services/AuthService';
import { PrinterService } from '../services/PrinterService';
import { FileService } from '../services/FileService';
import { Logger } from '../services/Logger';
import { IProcoreState } from './procore/state';
import { IState as IProjectNewState } from './project-new/state';

export interface IAction {
  type: string;
  payload: any;
}

export interface IRootState {
  auth: IAuthState;
  user: IUserState;
  client: IClientState;
  project: IProjectState;
  accessControlSystem: IAccessControlSystemState;
  badgePrinterSystem: IBadgePrintingSystemState;
  worker: IWorkerState;
  certification: ICertificationState;
  training: ITrainingState;
  invoice: IInvoiceState;
  file: IFileState;
  badge: IBadgeState;
  statistics: IStatisticsState;
  payment: IPaymentState;
  general: IGeneralState;
  router?: { location: { pathname: string; search: string; hard: string; key: string }; action: string };
  procore: IProcoreState;
  projectNew: IProjectNewState;
}

export interface IEpicDependencies {
  history: History;
  apiService: ApiService;
  authService: AuthService;
  printerService: PrinterService;
  fileService: FileService;
  logger: typeof Logger;
}
