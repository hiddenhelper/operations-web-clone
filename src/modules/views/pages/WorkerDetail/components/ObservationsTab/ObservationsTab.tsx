import React, { useEffect, memo, useMemo, useState, useCallback } from 'react';
import {
  Button,
  Table,
  TableHead,
  TableCell,
  TableRow,
  TableBody,
  Box,
  Divider,
  DialogActions,
  DialogTitle,
  DialogContent,
  Typography,
  withStyles,
} from '@material-ui/core';

import EmptyList from 'modules/views/shared/EmptyList';
import Pagination from 'modules/views/shared/Pagination';
import SelectFilter from 'modules/views/shared/SelectFilter';
import FileRow from 'modules/views/shared/FormHandler/FileUpload/FileRow';
import TableCellLink from 'modules/views/shared/TableCellLink';
import Modal from 'modules/views/shared/Modal';
import WorkerProjectFilter from '../WorkerProjectFilter';

import { WorkerModel, ProjectModel, GeneralModel, FileModel } from 'modules/models';
import { WorkersIcon } from 'constants/index';
import { getDefaultValue, getFormattedDate } from 'utils';
import { listTableRowStyles, tableGlobalStyles } from 'assets/styles';
import { useStyles as modalStyles } from '../../../../shared/Modal/style';
import { useStyles as buttonStyles } from '../../../../shared/FormHandler/ControlledButton/styles';

export interface IObservationsTabProps {
  worker: WorkerModel.IWorker;
  workerObservationList: WorkerModel.IWorkerObservation[];
  projectMap: GeneralModel.IEntityMap<GeneralModel.IEntityMap<ProjectModel.IWorkerProject>>;
  queryParams: GeneralModel.IQueryParams;
  workerObservationCount: number;
  observationDetail: WorkerModel.IWorkerObservation;
  detailLoading: GeneralModel.ILoadingStatus;
  fetchWorkerObservations: (id: string, query: GeneralModel.IQueryParams) => void;
  fetchWorkerProjectList: (id: string, query: GeneralModel.IQueryParams) => void;
  fetchObservationDetail: (workerId: string, id: string) => void;
  onPageChange: (page: number) => void;
  onFilterPeriodChange: (period: number) => void;
  onFilterProjectChange: (projectId: string) => void;
}
const StyledTableRow = withStyles(listTableRowStyles)(TableRow);

const ObservationsTab = ({
  worker,
  queryParams,
  workerObservationList,
  projectMap,
  workerObservationCount,
  observationDetail,
  detailLoading,
  fetchWorkerProjectList,
  fetchWorkerObservations,
  fetchObservationDetail,
  onPageChange,
  onFilterPeriodChange,
  onFilterProjectChange,
}: IObservationsTabProps) => {
  const tableGlobalClasses = tableGlobalStyles();
  const modalClasses = modalStyles();
  const buttonClasses = buttonStyles();
  const pageCount = useMemo(() => Math.ceil(workerObservationCount / queryParams.limit), [workerObservationCount, queryParams.limit]);
  const [detailModal, setDetailModal] = useState<{ isOpen: boolean; id: string }>({ isOpen: false, id: null });

  const handleCloseModal = useCallback(() => {
    setDetailModal({ isOpen: false, id: null });
  }, [setDetailModal]);

  const handleSelectObservation = useCallback(
    (id: string) => {
      setDetailModal({ isOpen: true, id });
    },
    [setDetailModal]
  );

  useEffect(() => {
    fetchWorkerObservations(worker?.id, queryParams);
  }, [worker, queryParams, fetchWorkerObservations]);

  useEffect(() => {
    if (detailModal.id) fetchObservationDetail(worker?.id, detailModal.id);
  }, [detailModal.id, fetchObservationDetail, worker]);

  return (
    <>
      <div className={`${tableGlobalClasses.filterActionsContainer} ${tableGlobalClasses.filterActionsContainerPadding}`}>
        <div className={tableGlobalClasses.filterActionsContainerLeft}>
          <WorkerProjectFilter
            workerId={worker?.id}
            queryParams={queryParams}
            projectMap={projectMap[worker?.id]}
            onFilterProjectChange={onFilterProjectChange}
            fetchWorkerProjectList={fetchWorkerProjectList}
          />
          <Divider className={tableGlobalClasses.dividerColor} orientation="vertical" flexItem={true} />
          <Box className={`${tableGlobalClasses.filterStatusContainer} ${tableGlobalClasses.leftFilterStatusContainer}`}>
            <SelectFilter
              value={getDefaultValue(GeneralModel.timeFilterTypeMap[queryParams.period], 'All Times')}
              optionList={GeneralModel.workerActivityPeriodList}
              onChange={onFilterPeriodChange}
            />
          </Box>
        </div>
      </div>
      {!workerObservationList.length && <EmptyList icon={<WorkersIcon />} text="There are no Observations" />}
      {!!workerObservationList.length && (
        <>
          <Table aria-label="project-list">
            <TableHead>
              <TableRow>
                <TableCell>Type</TableCell>
                <TableCell>Date</TableCell>
                <TableCell>Project</TableCell>
                <TableCell>Standard Description</TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {workerObservationList.map((observation, index) => (
                <StyledTableRow data-testid="observation-list-row" key={index}>
                  <TableCell
                    data-testid="observation-row-type"
                    className={tableGlobalClasses.buttonCell}
                    onClick={() => handleSelectObservation(observation.id)}
                  >
                    {observation.type?.name}
                  </TableCell>
                  <TableCell>{getFormattedDate(observation.date, GeneralModel.DateFormat.DATE_TIME)}</TableCell>
                  <TableCell>
                    {observation?.project?.id ? (
                      <TableCellLink href={`/projects/detail/${observation.project.id}`} text={observation.project.name} title="View Project detail" />
                    ) : (
                      '-'
                    )}
                  </TableCell>
                  <TableCell>{observation.standardDescription?.name}</TableCell>
                </StyledTableRow>
              ))}
            </TableBody>
          </Table>
          <Pagination data-testid="pagination" page={queryParams.page} count={pageCount} onChange={onPageChange} />
        </>
      )}
      <Modal
        show={detailModal.isOpen}
        onClose={handleCloseModal}
        styleClass={`${modalClasses.dialogContainer} ${modalClasses.detailModal}`}
        render={() =>
          detailLoading && detailLoading.isLoading ? (
            'Loading...'
          ) : (
            <>
              <DialogTitle data-testid="alert-dialog" id="alert-dialog-title">
                Worker Observation
              </DialogTitle>
              <DialogContent>
                <Typography variant="subtitle1">{`${getFormattedDate(observationDetail?.date, GeneralModel.DateFormat.DATE)}. ${
                  observationDetail?.company?.name
                }. ${observationDetail?.project?.name}`}</Typography>
                <Typography variant="h6" className={modalClasses.modalSubtitle}>{`Reported by: ${getDefaultValue(observationDetail?.owner?.name)}`}</Typography>
                <Typography variant="h6" className={modalClasses.modalSubtitle}>
                  {`${observationDetail?.type?.name}: ${observationDetail?.standardDescription?.name}`}
                </Typography>
                <Typography variant="body1" className={modalClasses.modalDescription}>
                  {getDefaultValue(observationDetail?.comments, '')}
                </Typography>
                {observationDetail?.files.length > 0 && (
                  <div className={modalClasses.fileListWrapper}>
                    {observationDetail?.files.map((file, index) => (
                      <div key={index} style={{ marginBottom: '5px' }}>
                        <FileRow
                          file={FileModel.getFallbackFile({ name: file.displayName, size: file.fileSize, url: file.url })}
                          showProgress={false}
                          showDelete={false}
                          showInModal={true}
                          allowDownload={true}
                        />
                      </div>
                    ))}
                  </div>
                )}
              </DialogContent>
              <DialogActions>
                <Button
                  className={`${buttonClasses.closeButtonWidth} ${buttonClasses.saveButton}`}
                  color="primary"
                  variant="contained"
                  autoFocus={true}
                  onClick={handleCloseModal}
                  data-testid="detail-modal-close-btn"
                >
                  Close
                </Button>
              </DialogActions>
            </>
          )
        }
      />
    </>
  );
};

export default memo(ObservationsTab);
