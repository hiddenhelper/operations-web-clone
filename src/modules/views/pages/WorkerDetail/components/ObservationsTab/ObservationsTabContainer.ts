import { Dispatch } from 'redux';
import { connect } from 'react-redux';

import ObservationsTab from './ObservationsTab';

import { GeneralModel } from '../../../../../models';
import { IRootState } from '../../../../../state-mgmt/rootState';
import { projectState } from '../../../../../state-mgmt/project';
import { workerState } from '../../../../../state-mgmt/worker';
import { GENERAL } from '../../../../../../constants';

export const mapStateToProps = (state: IRootState) => ({
  workerObservationList: state.worker.workerObservationList,
  projectMap: state.project.projectWorkerMap,
  workerObservationCount: state.worker.workerObservationCount,
  observationDetail: state.worker.observation,
  detailLoading: state.general.loadingMap[GENERAL.LOADING_KEY.FETCH_WORKER_OBSERVATION],
});

export const mapDispatchToProps = (dispatch: Dispatch) => ({
  fetchWorkerObservations: (id: string, query: GeneralModel.IQueryParams) => dispatch(workerState.actions.fetchWorkerObservationListStart(id, query)),
  fetchWorkerProjectList: (id: string, query: GeneralModel.IQueryParams) => dispatch(projectState.actions.fetchWorkerProjectListStart(id, query)),
  fetchObservationDetail: (workerId: string, id: string) => dispatch(workerState.actions.fetchWorkerObservationStart(workerId, id)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ObservationsTab);
