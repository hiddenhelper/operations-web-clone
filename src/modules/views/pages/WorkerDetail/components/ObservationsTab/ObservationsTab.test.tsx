import React from 'react';

import { MemoryRouter } from 'react-router';
import { render, act, fireEvent } from '@testing-library/react';

import ObservationsTab, { IObservationsTabProps } from './ObservationsTab';
import { getWorker_1, getWorkerProject_1, getWorkerObservation_1, getDefaultLoading } from '../../../../../../test/entities';
import { createMockStore } from 'redux-test-utils';
import { getInitialState } from '../../../../../../test/rootState';
import { Provider } from 'react-redux';

describe.skip('ObservationsTab', () => {
  let props: IObservationsTabProps;

  beforeEach(() => {
    props = {
      projectMap: { [getWorker_1().id]: { [getWorkerProject_1().id]: getWorkerProject_1() } },
      queryParams: {},
      observationDetail: getWorkerObservation_1(),
      worker: getWorker_1(),
      detailLoading: getDefaultLoading(),
      workerObservationList: [getWorkerObservation_1()],
      workerObservationCount: 1,
      fetchWorkerObservations: jest.fn(),
      fetchWorkerProjectList: jest.fn(),
      onPageChange: jest.fn(),
      onFilterPeriodChange: jest.fn(),
      onFilterProjectChange: jest.fn(),
      fetchObservationDetail: jest.fn(),
    };
  });

  it.skip('should render', () => {
    const { container } = render(
      <MemoryRouter>
        <Provider store={createMockStore(getInitialState()) as any}>
          <ObservationsTab {...props} />
        </Provider>
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render empty list', () => {
    props.workerObservationList = [];
    const { container } = render(
      <MemoryRouter>
        <Provider store={createMockStore(getInitialState()) as any}>
          <ObservationsTab {...props} />
        </Provider>
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  it('should open modal', () => {
    const wrapper = render(
      <MemoryRouter>
        <Provider store={createMockStore(getInitialState()) as any}>
          <ObservationsTab {...props} />
        </Provider>
      </MemoryRouter>
    );

    act(() => {
      fireEvent.click(wrapper.getByTestId('observation-row-type'));
    });

    act(() => {
      fireEvent.click(wrapper.getByTestId('detail-modal-close-btn'));
    });

    expect(props.fetchObservationDetail).toHaveBeenCalled();
  });

  it.skip('should show detail loading', () => {
    props.detailLoading = { ...getDefaultLoading(), isLoading: true };
    const wrapper = render(
      <MemoryRouter>
        <Provider store={createMockStore(getInitialState()) as any}>
          <ObservationsTab {...props} />
        </Provider>
      </MemoryRouter>
    );

    act(() => {
      fireEvent.click(wrapper.getByTestId('observation-row-type'));
    });

    expect(wrapper.container).toMatchSnapshot();
  });
});
