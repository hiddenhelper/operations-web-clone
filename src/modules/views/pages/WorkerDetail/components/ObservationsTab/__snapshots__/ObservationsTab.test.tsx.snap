// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObservationsTab should render 1`] = `
<div>
  <div
    class="makeStyles-filterActionsContainer-12 makeStyles-filterActionsContainerPadding-14"
  >
    <div
      class="makeStyles-filterActionsContainerLeft-13"
    >
      <div
        class="MuiBox-root MuiBox-root-92 makeStyles-filterStatusContainer-15 makeStyles-autocompleteFilterStatus-32"
      >
        <div
          class="makeStyles-floatingFilterWrapper-21 makeStyles-filterWrapper-20 makeStyles-buttonContainer-94 floatingFilterWrapper"
        >
          <span
            class="makeStyles-floatingFilterButton-22"
            data-testid="controlled-button"
            tabindex="-1"
          >
            <button
              aria-controls="false"
              aria-haspopup="true"
              class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttonFilter-30"
              data-testid="button-filter-open"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                All Projects
                <span
                  class="MuiButton-endIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    class="makeStyles-dropdownIcon-24"
                  >
                    expand.svg
                  </svg>
                </span>
              </span>
            </button>
          </span>
        </div>
      </div>
      <hr
        class="MuiDivider-root makeStyles-dividerColor-25 MuiDivider-flexItem MuiDivider-vertical"
      />
      <div
        class="MuiBox-root MuiBox-root-98 makeStyles-filterStatusContainer-15 makeStyles-leftFilterStatusContainer-19"
      >
        <div
          class="makeStyles-floatingFilterWrapper-21 makeStyles-filterWrapper-20  floatingFilterWrapper"
        >
          <span
            class="makeStyles-floatingFilterButton-22"
            data-testid="controlled-button"
            tabindex="-1"
          >
            <button
              aria-controls="false"
              aria-haspopup="true"
              class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttonFilter-30"
              data-testid="button-filter-open"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                All Times
                <span
                  class="MuiButton-endIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    class="makeStyles-dropdownIcon-24"
                  >
                    expand.svg
                  </svg>
                </span>
              </span>
            </button>
          </span>
        </div>
      </div>
    </div>
  </div>
  <table
    aria-label="project-list"
    class="MuiTable-root"
  >
    <thead
      class="MuiTableHead-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Type
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Date
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Project
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Standard Description
        </th>
      </tr>
    </thead>
    <tbody
      class="MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-99"
        data-testid="observation-list-row"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body makeStyles-buttonCell-38"
          data-testid="observation-row-type"
        >
          Ladder
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          Dec 15, 2020 07:00 pm
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          <a
            class="Link-root-100"
            href="/projects/detail/1"
            title="View Project detail"
          >
            prj1
          </a>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          Not using the ladder properly
        </td>
      </tr>
    </tbody>
  </table>
  <nav
    aria-label="pagination navigation"
    class="MuiPagination-root  makeStyles-pagination-101"
    data-testid="pagination"
  >
    <ul
      class="MuiPagination-ul"
    >
      <li>
        <button
          aria-label="Go to previous page"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-disabled Mui-disabled"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiPaginationItem-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-label="Go to next page"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiPaginationItem-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`ObservationsTab should render empty list 1`] = `
<div>
  <div
    class="makeStyles-filterActionsContainer-113 makeStyles-filterActionsContainerPadding-115"
  >
    <div
      class="makeStyles-filterActionsContainerLeft-114"
    >
      <div
        class="MuiBox-root MuiBox-root-193 makeStyles-filterStatusContainer-116 makeStyles-autocompleteFilterStatus-133"
      >
        <div
          class="makeStyles-floatingFilterWrapper-122 makeStyles-filterWrapper-121 makeStyles-buttonContainer-195 floatingFilterWrapper"
        >
          <span
            class="makeStyles-floatingFilterButton-123"
            data-testid="controlled-button"
            tabindex="-1"
          >
            <button
              aria-controls="false"
              aria-haspopup="true"
              class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttonFilter-131"
              data-testid="button-filter-open"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                All Projects
                <span
                  class="MuiButton-endIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    class="makeStyles-dropdownIcon-125"
                  >
                    expand.svg
                  </svg>
                </span>
              </span>
            </button>
          </span>
        </div>
      </div>
      <hr
        class="MuiDivider-root makeStyles-dividerColor-126 MuiDivider-flexItem MuiDivider-vertical"
      />
      <div
        class="MuiBox-root MuiBox-root-199 makeStyles-filterStatusContainer-116 makeStyles-leftFilterStatusContainer-120"
      >
        <div
          class="makeStyles-floatingFilterWrapper-122 makeStyles-filterWrapper-121  floatingFilterWrapper"
        >
          <span
            class="makeStyles-floatingFilterButton-123"
            data-testid="controlled-button"
            tabindex="-1"
          >
            <button
              aria-controls="false"
              aria-haspopup="true"
              class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttonFilter-131"
              data-testid="button-filter-open"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                All Times
                <span
                  class="MuiButton-endIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    class="makeStyles-dropdownIcon-125"
                  >
                    expand.svg
                  </svg>
                </span>
              </span>
            </button>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class=" makeStyles-container-200"
  >
    <div>
      <div
        class="MuiAvatar-root MuiAvatar-circle makeStyles-missingAvatarWrapper-211 makeStyles-missingAvatarBig-212 MuiAvatar-colorDefault"
      >
        <svg>
          workers.svg
        </svg>
      </div>
    </div>
    <p
      class="MuiTypography-root makeStyles-missingListText-201 MuiTypography-body1"
    >
      There are no Observations
    </p>
  </div>
</div>
`;

exports[`ObservationsTab should show detail loading 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="makeStyles-filterActionsContainer-356 makeStyles-filterActionsContainerPadding-358"
  >
    <div
      class="makeStyles-filterActionsContainerLeft-357"
    >
      <div
        class="MuiBox-root MuiBox-root-436 makeStyles-filterStatusContainer-359 makeStyles-autocompleteFilterStatus-376"
      >
        <div
          class="makeStyles-floatingFilterWrapper-365 makeStyles-filterWrapper-364 makeStyles-buttonContainer-438 floatingFilterWrapper"
        >
          <span
            class="makeStyles-floatingFilterButton-366"
            data-testid="controlled-button"
            tabindex="-1"
          >
            <button
              aria-controls="false"
              aria-haspopup="true"
              class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttonFilter-374"
              data-testid="button-filter-open"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                All Projects
                <span
                  class="MuiButton-endIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    class="makeStyles-dropdownIcon-368"
                  >
                    expand.svg
                  </svg>
                </span>
              </span>
            </button>
          </span>
        </div>
      </div>
      <hr
        class="MuiDivider-root makeStyles-dividerColor-369 MuiDivider-flexItem MuiDivider-vertical"
      />
      <div
        class="MuiBox-root MuiBox-root-442 makeStyles-filterStatusContainer-359 makeStyles-leftFilterStatusContainer-363"
      >
        <div
          class="makeStyles-floatingFilterWrapper-365 makeStyles-filterWrapper-364  floatingFilterWrapper"
        >
          <span
            class="makeStyles-floatingFilterButton-366"
            data-testid="controlled-button"
            tabindex="-1"
          >
            <button
              aria-controls="false"
              aria-haspopup="true"
              class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttonFilter-374"
              data-testid="button-filter-open"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                All Times
                <span
                  class="MuiButton-endIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    class="makeStyles-dropdownIcon-368"
                  >
                    expand.svg
                  </svg>
                </span>
              </span>
            </button>
          </span>
        </div>
      </div>
    </div>
  </div>
  <table
    aria-label="project-list"
    class="MuiTable-root"
  >
    <thead
      class="MuiTableHead-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Type
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Date
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Project
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Standard Description
        </th>
      </tr>
    </thead>
    <tbody
      class="MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-443"
        data-testid="observation-list-row"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body makeStyles-buttonCell-382"
          data-testid="observation-row-type"
        >
          Ladder
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          Dec 15, 2020 07:00 pm
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          <a
            class="Link-root-444"
            href="/projects/detail/1"
            title="View Project detail"
          >
            prj1
          </a>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          Not using the ladder properly
        </td>
      </tr>
    </tbody>
  </table>
  <nav
    aria-label="pagination navigation"
    class="MuiPagination-root  makeStyles-pagination-445"
    data-testid="pagination"
  >
    <ul
      class="MuiPagination-ul"
    >
      <li>
        <button
          aria-label="Go to previous page"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-disabled Mui-disabled"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiPaginationItem-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-label="Go to next page"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiPaginationItem-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
    </ul>
  </nav>
</div>
`;
